ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/OnDisk.hs

```diff
@@ -681,8 +698,10 @@ runDB standalone@DB{..} cmd =
                           return $ MaybeErr (Right ())
 
     truncateSnapshot :: SomeHasFS m -> DiskSnapshot -> m ()
-    truncateSnapshot (SomeHasFS hasFS@HasFS{..}) ss =
-        withFile hasFS (snapshotToDirPath ss) (AppendMode AllowExisting) $ \h ->
+    truncateSnapshot (SomeHasFS hasFS@HasFS{..}) ss = do
+        withFile hasFS (snapshotToStatePath ss) (AppendMode AllowExisting) $ \h ->
```

Comment:
Why do we have the nested `withFile` now?

Reply:
It's not nested; it's two different files. (The new one is the file that the in-memory backing store writes when we take a snapshot.)

Reply:
Oh I see, sorry I misread this then.